weave-js/src/components/PagePanelComponents/Home/Browse3/pages/LeaderboardPage/SimpleCodeLikeTextArea.tsx

```diff
@@ -0,0 +1,221 @@
+/**
+ * A simple multi-line text editor component that mimics code editor styling.
+ * Features auto-sizing, manual resize handle, and code-like formatting.
+ *
+ * Inspired by: weave-js/src/components/Form/TextField.tsx
+ */
+
+import {Tailwind} from '@wandb/weave/components/Tailwind';
+import classNames from 'classnames';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+
+export const TextAreaSizes = {
+  Medium: 'medium',
+  Large: 'large',
+} as const;
+export type TextAreaSize = (typeof TextAreaSizes)[keyof typeof TextAreaSizes];
+
+type TextAreaProps = {
+  placeholder?: string;
+  value?: string;
+  onChange?: (value: string) => void;
+  onKeyDown?: (
+    key: string,
+    e: React.KeyboardEvent<HTMLTextAreaElement>
+  ) => void;
+  onBlur?: (value: string) => void;
+  autoFocus?: boolean;
+  disabled?: boolean;
+  ariaLabel?: string;
+  errorState?: boolean;
+  maxLength?: number;
+  maxRows?: number;
+  dataTest?: string;
+};
+
+export const SimpleCodeLikeTextArea = ({
+  placeholder,
+  value,
+  onChange,
+  onKeyDown,
+  onBlur,
+  autoFocus,
+  disabled,
+  ariaLabel,
+  errorState,
+  maxLength,
+  maxRows = 8,
+  dataTest,
+}: TextAreaProps) => {
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const [isManuallyResized, setIsManuallyResized] = useState(false);
+  const isDraggingRef = useRef(false);
+  const initialHeightRef = useRef<number>(0);
+  const initialMouseYRef = useRef<number>(0);
+
+  // Automatically adjust height based on content
+  const adjustHeight = () => {
+    const textarea = textareaRef.current;
+    if (!textarea || isManuallyResized) {
+      return;
+    }
+
+    textarea.style.height = 'auto';
+    const lineHeight = parseInt(
+      getComputedStyle(textarea).lineHeight || '20',
```

Comment:
wow `getComputedStyle(textarea).lineHeight` is a string??

Reply:
<img width="722" alt="Screenshot 2024-10-31 at 12 47 49" src="https://github.com/user-attachments/assets/405f49fd-66b7-4611-9fb5-df0c22f81b9a">
 Indeed it is!